sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +383,512 @@ def transform(self, X):
+
+    n_imputations : int, optional (default=100)
+        Number of MICE rounds to perform the results of which will be
+        used in the final average.
+
+    n_burn_in : int, optional (default=10)
+        Number of initial MICE rounds to perform the results of which
+        will not be returned.
+
+    predictor : estimator object, default=BayesianRidge
+        The predictor to use at each step of the round-robin imputation.
+        As of now, it must support `return_std` in its ``predict`` method.
+
+    n_nearest_features : int, optional (default=None)
+        Number of other features to use to estimate the missing values of
+        the each feature column. Nearness between features is measured using
+        the absolute correlation coefficient. Can provide significant
+        speed-up when the number of features is huge. If `None`, all features
+        will be used.
+
+    initial_strategy : str, optional (default="mean")
+        Which strategy to use to initialize the missing values. Same as the
+        `strategy` option in :class:`sklearn.preprocessing.Imputer.`
+        Valid values: {"mean", "median", or "most_frequent"}.
+
+    min_value : float, optional (default=None)
+        Minimum possible imputed value.
+
+    max_value : float, optional (default=None)
+        Maximum possible imputed value.
+
+    verbose : boolean, optional (default=False)
+        Controls the verbosity of the imputer.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        The seed of the pseudo random number generator to use when shuffling
+        the data.  If int, random_state is the seed used by the random number
+        generator; If RandomState instance, random_state is the random number
+        generator; If None, the random number generator is the RandomState
+        instance used by `np.random`.
+
+    Attributes
+    ----------
+    initial_imputer_ : object of class :class:`sklearn.preprocessing.Imputer`'
+        The imputer used to initialize the missing values.
+
+    imputation_sequence_ : list of tuples
+        Each tuple has `(feat_idx, neighbor_feat_idx, predictor)`, where
+        `feat_idx` is the current feature to be imputed, `neighbor_feat_idx`
+        is the list of other features used to impute the current feature, and
+        `predictor` is the trained predictor used for the imputation.
+
+    Notes
+    -----
+    Features which contain only missing values at ``fit`` are discarded upon
+    ``transform``.
+
+    Features with missing values in transform which did not have any missing
+    values in fit will be imputed with the initial imputation method.
+
+    References
+    ----------
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
```

Comment:
We would usually rather put self on the first line, and indent remaining lines to the same character as self starts on...